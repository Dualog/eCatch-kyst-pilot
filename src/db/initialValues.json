{
  "isOffline": false,
  "isDarkMode": false,
  "dialog": {"open": false},
  "notification": {"open": false},
  "messages": [],
  "trips": [],
  "isLoading": true,
  "DCAStarted": false,
  "fishOnBoard": {},
  "errors": {},
  "fields": {
    "departure": null,
    "PO": null,
    "expectedFishingStart": "",
    "DS": "",
    "OB": {},
    "KG": {},
    "CA": {},
    "expectedFishingSpot": {},
    "startFishingSpot": {},
    "endFishingSpot": {},
    "QI": 1,
    "AC": "FIS",
    "GP": 0,
    "ZO": "NOR",
    "DU": 1
  },
  "custom": {
    "editing": {},
    "fishingSpots": [],
    "ports": [],
    "fishingGear": [],
    "activity": [],
    "species": [],
    "fishingPermit": [],
    "ZO": []
  }
}

