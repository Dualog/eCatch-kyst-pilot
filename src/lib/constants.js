export const fields = {
  common: ["TM", "RN", "RC", "MA", "DA", "TI"],
  DEP: ["PO", "ZD", "ZT", "OB", "PD", "PT", "LA", "LO", "AC", "DS"],
  // NOTE: what is TS?
  DCA: [/*"MV",*/ "AD", "NA", "XR", "QI", "AC", /*"TS",*/ "BD","BT", "ZO", "LT", "LG", "GE", "GP", "XT", "XG", "DU", "CA", /*"ME", "GS" These are dependent on GE,*/],
  POR: ["AD", "NA", "XR", "PO", "PD", "PT", "OB", "LS", "KG"]
}

export const VALIDTM = ["DEP", "DCA", "POR"]
