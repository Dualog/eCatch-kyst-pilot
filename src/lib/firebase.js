import firebase from "@firebase/app"
import "@firebase/database"
import "@firebase/firestore"
import "@firebase/auth"
import "@firebase/storage"


const config = {
  apiKey: "AIzaSyD5l09unKGP3D3cFY_mr_BQPgcTjhkU2jc",
  authDomain: "ecatch-kyst.firebaseapp.com",
  databaseURL: "https://ecatch-kyst.firebaseio.com",
  projectId: "ecatch-kyst",
  storageBucket: "ecatch-kyst.appspot.com",
  messagingSenderId: "1037945604073"
}

export default !firebase.apps.length ? firebase.initializeApp(config) : firebase.app()

export const FS = firebase.firestore()
FS.enablePersistence()
  .catch(({code}) => {
    switch (code) {
    case 'failed-precondition':
      // Multiple tabs open, persistence can only be enabled
      // in one tab at a time.
      break
    case 'unimplemented':
      // The current browser does not support all of the
      // features required to enable persistence
      break
    default:
      break
    }
  })
export const DB = firebase.database()
export const FileStore = firebase.storage()
export const AUTH = firebase.auth()

export const TIMESTAMP_SERVER = firebase.firestore.FieldValue.serverTimestamp()
export const TIMESTAMP_CLIENT = firebase.firestore.Timestamp.now
export const GEOPOINT = (lat, long) => new firebase.firestore.GeoPoint(lat, long)

export const CONNECTION_REF = DB.ref(".info/connected")
export const USERS_FS = FS.collection("users")