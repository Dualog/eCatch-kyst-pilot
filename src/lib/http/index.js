import * as functions from "firebase-functions"

export const getMessages = functions.https.onRequest(async (req, res) => {
  // Get messages
  //const messages = await BOATS_FS.where("length", ">", 10).get().data()
  // return messages

  //res.send(JSON.stringify(messages))
  res.send("Hello!")
})
