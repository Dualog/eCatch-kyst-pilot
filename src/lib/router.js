export const routes = {
  ROOT: "/",
  DASHBOARD: "/dashboard",
  TEST: "/test",
  FORGOT_PASSWORD: "/new-password",
  REGISTER: "/register",
  PROFILE: "/profile",
  TRIPS: "/trips",
  MESSAGES: "/messages",
  DEP: "/DEP",
  DCA0: "/DCA0",
  DCA: "/DCA",
  POR: "/POR",
  NOT_FOUND: "/404",
  NEW: "/new",
  EDIT: "/edit"
}