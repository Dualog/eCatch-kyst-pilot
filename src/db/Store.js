import React, {Component, createContext} from "react"
import initValues from "./initialValues.json"
import {CONNECTION_REF} from "../lib/firebase"

import * as darkMode from "./actions/darkMode"
import {login, updateProfile, logout, deleteUser} from "./actions/users"

const Store = createContext()

export class Database extends Component {

  state = initValues

  async componentDidMount() {

    this.initDarkMode()

    this.userLogin()

    CONNECTION_REF
      .on("value", snap => this.setState({isOffline: !snap.val()}))

  }

  // Dark mode
  initDarkMode = darkMode.init.bind(this)

  toggleDarkMode = darkMode.toggle.bind(this)


  // User
  userLogin = login.bind(this)

  userLogout = logout.bind(this)

  userDelete = deleteUser.bind(this)

  userUpdateProfile = updateProfile.bind(this)

  render() {
    return (
      <Store.Provider
        value={{
          handleToggleDarkMode: this.toggleDarkMode,
          handleUserUpdateProfile: this.userUpdateProfile,
          handleUserLogout: this.userLogout,
          handleUserLogin: this.userLogin,
          handleUserDelete: this.userDelete,
          ...this.state
        }}
      >
        {this.props.children}
      </Store.Provider>
    )
  }
}

export default Store