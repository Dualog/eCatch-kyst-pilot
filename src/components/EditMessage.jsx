import React from 'react'

import {Redirect, withRouter} from "react-router-dom"
import {Card, CardContent} from '@material-ui/core'
import {routes} from '../lib/router.js'
import {withStore} from '../db'
import Form from './Forms/Form'
import {isBefore, addHours} from 'date-fns'

export default withStore(withRouter(({store: {messages}, match: {params: {messageId}}}) => {

  const {created} = messages.find(message => message.RN === parseInt(messageId, 10)) || {}
  const disabled = created && isBefore(addHours(created.toDate(), 12), Date.now())
  return (
    <Card>
      {disabled ? <Redirect to={routes.MESSAGES}/>: null}
      <CardContent>
        <Form messageId={messageId} type="DCA"/>
      </CardContent>
    </Card>
  )
}))