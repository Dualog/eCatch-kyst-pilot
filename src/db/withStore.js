import React from 'react'
import Store from '.'

/**
 * HOC implementation of Store context
 * @param {Component} WrappedComponent The component to pass the store values to
 * @returns {Component} new component with the Store values in this.props.store
 */
const withStore = WrappedComponent => props =>
  <Store.Consumer>
    {store => <WrappedComponent store={store} {...props }/>}
  </Store.Consumer>

export default withStore