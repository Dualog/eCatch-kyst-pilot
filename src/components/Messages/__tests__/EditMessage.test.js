import EditMessage from '../EditMessage'
import {Redirect} from "react-router-dom"
import "../../../db"
import {routes} from "../../../lib/router"

jest.mock("../../../db", () => ({
  withStore: Component => props => <Component {...props}/>
}))

describe("EditMessage component", () => {
  const props = {
    store: {messages:{
      RN: jest.fn(),
      TM: "DCA",
      created: jest.fn()
    }},
    match: {params: {messageId: jest.fn()}}
  }
  const wrapper = shallow(<EditMessage {... props}/>)

  it("renders correctly", () => {
    expect(wrapper).toHaveLength(1)
  })

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  /*
  it('correct messageId', () => {
    wrapper.setProps({store:{messages:{RN: 1}}})
    wrapper.setProps({match:{params:{messageId:1}}})
    expect({store:{messages:{RN}}}).toBe({match:{params:{messageId}}})
  })*/
  /*
  it("if created is too old, redirect to MESSAGES", () => {
    wrapper.setProps({store:{messages:{created: new Date(2018, 6, 10, 23, 0)}}})
    expect(wrapper.find(Redirect)).toHaveLength(1) //this fails for now, the length is 0.
    expect(wrapper.find(Redirect).prop("to")).toBe(routes.MESSAGES)
    wrapper.setProps({store:{messages:{created: Date.now()}}})
  })*/

})

