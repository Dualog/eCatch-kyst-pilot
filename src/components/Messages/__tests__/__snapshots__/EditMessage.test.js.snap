// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMessage component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    match={
      Object {
        "params": Object {
          "messageId": [MockFunction],
        },
      }
    }
    store={
      Object {
        "messages": Object {
          "RN": [MockFunction],
          "TM": "DCA",
          "created": [MockFunction],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "match": Object {
        "params": Object {
          "messageId": [MockFunction],
        },
      },
      "store": Object {
        "messages": Object {
          "RN": [MockFunction],
          "TM": "DCA",
          "created": [MockFunction],
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "match": Object {
          "params": Object {
            "messageId": [MockFunction],
          },
        },
        "store": Object {
          "messages": Object {
            "RN": [MockFunction],
            "TM": "DCA",
            "created": [MockFunction],
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
